src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -218,4 +220,31 @@ private void validateClusteringOrder(@Nullable Schema schema) {
       }
     }
   }
+
+  private void validateOperationProperties(@Nullable Schema schema) {
+    if (Arrays.stream(Operation.values()).map(Enum::name).noneMatch(operation.toUpperCase()::equals)) {
+      throw new InvalidConfigPropertyException(String.format("'%s' is incorrect value for field 'Operation'. " +
+                                                               "This field should contain one of the next values: " +
+                                                               "INSERT, UPDATE or UPSERT.", operation), NAME_OPERATION);
+    }
+    if (!Operation.INSERT.equals(getOperation()) && getRelationTableKey() == null) {
+      throw new InvalidConfigPropertyException(
+        "Table key value should be set in case UPDATE or UPSERT operations selected.", NAME_TABLE_KEY);
+    }
+    if (schema == null) {
+      return;
+    }
+    List<String> fields = Objects.requireNonNull(schema.getFields()).stream().map(Schema.Field::getName)
+      .collect(Collectors.toList());
+    List<String> keyFields = Arrays.stream(Objects.requireNonNull(getRelationTableKey()).split(","))
+      .map(String::trim).collect(Collectors.toList());
+    if (!Operation.INSERT.equals(getOperation()) &&  keyFields.size() >= fields.size()) {
+      throw new InvalidConfigPropertyException(
+        "Count of Table key value should be less then table columns count.", NAME_TABLE_KEY);
```

Comment:
this is not a useful error message. Should validate that all fields in the table key are also in the schema. If not, the error message should say something like:

Fields x, y, and z are in the table key, but not in the input schema.

Reply:
Done.